zerver/tests/test_email_mirror.py

```diff
@@ -76,22 +76,13 @@ def test_encode_decode(self) -> None:
         self.assertEqual(token, stream.email_token)
 
         parts = email_address.split('@')
-        parts[0] += "+include-footer+show-sender+include-quotations"
+        # Use a mix of + and . as separators, to test that it works:
+        parts[0] += "+include-footer.show-sender+include-quotations"
         email_address_all_options = '@'.join(parts)
         token, options = decode_email_address(email_address_all_options)
         self._assert_options(options, show_sender=True, include_footer=True, include_quotations=True)
         self.assertEqual(token, stream.email_token)
 
-        email_address_dots = email_address.replace('+', '.')
-        token, options = decode_email_address(email_address_dots)
-        self._assert_options(options)
-        self.assertEqual(token, stream.email_token)
-
-        email_address_dots_all_options = email_address_all_options.replace('+', '.')
-        token, options = decode_email_address(email_address_dots_all_options)
-        self._assert_options(options, show_sender=True, include_footer=True, include_quotations=True)
-        self.assertEqual(token, stream.email_token)
-
```

Comment:
Hmm, now that I've merged this, looking at our tests -- I think we maybe don't want to have deleted these?  I do think it's important that our unit tests protect against the possibility that we in the future accidentally break the legacy email addresses advertised to users before this change.

Reply:
@timabbott  Hmm, that was my intention behind having a mix of `` + `` and `` . `` in the test:
```
        parts[0] += "+include-footer.show-sender+include-quotes"
        email_address_all_options = '@'.join(parts)
        token, options = decode_email_address(email_address_all_options)
        self._assert_options(options, show_sender=True, include_footer=True, include_quotes=True)
        self.assertEqual(token, stream.email_token)
```
as a protection making sure we accept both the + and . format, as well as mixes.
But if this isn't doesn't seem sufficient, we could add explicit testing of pure + and pure . formats as well :thinking:

Reply:
I mostly think it'd be good to have at least one example with a comment above it mentioning backwards-compatibility, as well as an example using the exact URLs that are displayed to users, and an example using all `+`s` for visual style.